### PR TITLE
append an abac rule for $KUBE_USER.

### LINKED ISSUE #28869: ABAC config does not respect $KUBE_USER
abac-authz-policy.jsonl hardcodes "admin" instead of using the $KUBE_USER variable. This means that "admin" is the only basic-auth username that will work with that abac config.

https://github.com/kubernetes/kubernetes/blob/master/cluster/saltbase/salt/kube-apiserver/abac-authz-policy.jsonl

### PR DESCRIPTION
**Pull request by cjcullen**
Allows the specified basic-auth credentials to be authorized against all resources.

Fixes #28869.

### CODE REVIEW

**Comment by cjcullen** [comment]
Doesn't look like this is correct on our debian-based scripts.

**Comment by adityakali** [comment]
I am not too familiar with salt, but the way I have seen this done in the past is to add a configurable variable in salt config, it your case, you can in https://github.com/kubernetes/kubernetes/blob/master/cluster/saltbase/salt/kube-apiserver/abac-authz-policy.jsonl something like:

{"apiVersion": "abac.authorization.kubernetes.io/v1beta1", "kind": "Policy", "spec": {"user":"{{kube_user}}", "namespace": "*", "resource": "*", "apiGroup": "*", "nonResourcePath": "*"}

and then
on debian: use saltstack to set the variable 
on gci: use 'sed' to substitute it with value of KUBE_USER

**Comment by cjcullen** [comment]
@adityakali Yeah you're right. I'll do this in the more "salty" way.

**Comment by cjcullen** [comment]
Okay, should be in a cleaner state now.

  - Added a line to abac-authz-policy.jsonl w/ the user specified by jinja variable `{{kube_user}}`
  - On Debian, configure-vm.sh puts `$KUBE_USER` into `grains.kube_user` for salt. Then the abac file is treated as a jinja template, so the value of `$KUBE_USER` gets into that line I added.
  - On GCI/Trusty, configure-helper.sh removes the jinja line, and replaces `{{kube_user}}` w/ the value of `$KUBE_USER` before copying the abac file.

PTAL

**Comment by adityakali** [comment]
LGTM

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -637,7 +637,12 @@ function start-kube-apiserver {
     webhook_config_volume="{\"name\": \"webhookconfigmount\",\"hostPath\": {\"path\": \"/etc/gcp_authz.config\"}},"
   fi
   local -r src_dir="${KUBE_HOME}/kube-manifests/kubernetes/gci-trusty"
-  cp "${src_dir}/abac-authz-policy.jsonl" /etc/srv/kubernetes/
+
+  local -r abac_policy_json="${src_dir}/abac-authz-policy.jsonl"
+  remove-salt-config-comments "${abac_policy_json}"
+  sed -i -e "s@{{kube_user}}@${KUBE_USER}@g" "${abac_policy_json}"
+  cp ${abac_policy_json} /etc/srv/kubernetes/
```
Comment (zmerlynn) [inline cluster/gce/gci/configure-helper.sh]: nit: (and below) quotes: `"${abac_policy_json}"`

**Comment by zmerlynn** [comment]
LGTM except nit. Fix and feel free to self-apply label.

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (cjcullen) [inline cluster/gce/gci/configure-helper.sh]: Done and done.

**Comment by cjcullen** [comment]
Change was just a rebase to get past the docs verification error from yesterday. Reapplying LGTM.

**Pull request merged**